dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-fee-record-correction-transient-form-data.controller/index.test.ts

```diff
@@ -0,0 +1,187 @@
+      // Arrange
+      mockFeeRecordExists.mockReturnValue(true);
+      mockTransientFormDataFind.mockResolvedValue(null);
+
+      // Act
+      await getFeeRecordCorrectionTransientFormData(req, res);
+
+      // Assert
+      expect(res._getStatusCode()).toEqual(HttpStatusCode.Ok);
+
+      const responseBody = res._getData() as GetFeeRecordCorrectionTransientFormDataResponse;
+      expect(responseBody).toEqual({});
+
+      expect(mockTransientFormDataFind).toHaveBeenCalledTimes(1);
+      expect(mockTransientFormDataFind).toHaveBeenCalledWith(tfmUserId, feeRecordId);
+    });
+
+    it(`should respond with a '${HttpStatusCode.NotFound}' if no fee record with the provided fee record id and report id is found`, async () => {
+      // Arrange
+      mockFeeRecordExists.mockReturnValue(false);
+
+      // Act
+      await getFeeRecordCorrectionTransientFormData(req, res);
+
+      // Assert
+      expect(res._getStatusCode()).toEqual(HttpStatusCode.NotFound);
+      expect(mockTransientFormDataFind).not.toHaveBeenCalled();
+    });
+
+    it("should respond with the specific error status if retrieving the transient form data throws an 'ApiError'", async () => {
+      // Arrange
+      const errorStatus = HttpStatusCode.NotFound;
+      mockFeeRecordExists.mockRejectedValue(new TestApiError(errorStatus, undefined));
+
+      // Act
+      await getFeeRecordCorrectionTransientFormData(req, res);
+
+      // Assert
+      expect(res._getStatusCode()).toEqual(errorStatus);
+    });
+
+    it("should respond with the specific error message if retrieving the transient form data throws an 'ApiError'", async () => {
+      // Arrange
+      const errorMessage = 'Some error message';
+      mockFeeRecordExists.mockRejectedValue(new TestApiError(undefined, errorMessage));
+
+      // Act
+      await getFeeRecordCorrectionTransientFormData(req, res);
+
+      // Assert
+      expect(res._getData()).toEqual(`Failed to get fee record correction transient form data: ${errorMessage}`);
+    });
+
+    it(`should respond with a '${HttpStatusCode.InternalServerError}' if an unknown error occurs`, async () => {
+      // Arrange
+      mockFeeRecordExists.mockRejectedValue(new Error('Some error'));
+
+      // Act
+      await getFeeRecordCorrectionTransientFormData(req, res);
+
+      // Assert
+      expect(res._getStatusCode()).toEqual(HttpStatusCode.InternalServerError);
+    });
+
+    it('should respond with a generic error message if an unknown error occurs', async () => {
+      // Arrange
+      mockFeeRecordExists.mockRejectedValue(new Error('Some error'));
+
+      // Act
+      await getFeeRecordCorrectionTransientFormData(req, res);
+
+      // Assert
+      expect(res._getData()).toEqual(`Failed to get fee record correction transient form data`);
+    });
+
+    it('should call fee record exists once', async () => {
```

Comment:
could these maybe go in a describe or have a more descriptive if? Bit confused since it's gone from error to calling fee record exists